app/models/miq_task.rb

```diff
@@ -295,6 +296,17 @@ def process_cancel
     end
   end
 
+  def attributes_hash_when_state_change(state, attributes = {})
```

Comment:
as discussed above - this method does look confusing, I will add direct update to appropriate methods.